src/components/Templates/NavBarTemplate/NavBar.component.tsx

```diff
@@ -1,10 +1,87 @@
+import IconButton from '@mui/material/IconButton';
+import Box from '@mui/material/Box';
+import Stack from '@mui/material/Stack';
+import Person2TwoToneIcon from '@mui/icons-material/Person2TwoTone';
+import MenuItem from '@mui/material/MenuItem';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import LogoutTwoToneIcon from '@mui/icons-material/LogoutTwoTone';
+import AccountCircleTwoToneIcon from '@mui/icons-material/AccountCircleTwoTone';
+import { useState } from 'react';
 import { Logo } from '../../Molucules/Logo';
 import { NavBar } from '../../Organisms/NavBar';
+import { ProfileMenu } from '../../Organisms/ProfileMenu';
+import { Link, NavLink, useNavigate } from 'react-router-dom';
+import { Button } from '@mui/material';
+import { useUser } from '../../../context/user';
 
 const NavBarTemplate = () => {
+  const navigate = useNavigate();
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const isProfileMenuOpen = Boolean(anchorEl);
+  const { signOut, user } = useUser();
```

Comment:
@siroshan Check all the files where this context is used and implement the change I've mentioned in a below comment!